### ISSUE TITLE
Seeing more downlink messages than expected in TTS

### ISSUE THREAD

**Issue by pjb304** [computenodes/dragino#19]
![image](https://user-images.githubusercontent.com/1146400/125088271-2cd0e980-e0c5-11eb-91de-df731cb1e8d0.png)
I haven't scheduled any downlink messages, so not sure where they are coming from.
Don't see this behaviour in v2
![image](https://user-images.githubusercontent.com/1146400/125088384-45410400-e0c5-11eb-83f6-f6a107f0ef56.png)

**Comment by BNNorman** [comment]
Don't you mean you haven't scheduled any uplink messages? The frame counter is showing zero down frames but 5 up. 

Or am I misunderstanding the up/down in that screen shot?

**Comment by BNNorman** [comment]
15 up, 14 down suggests you are sending Conf_data_up messages and getting the confirmation reply.

Btw, ttn fair use limits you to 10 downlinks per day.

**Comment by pjb304** [comment]
Agreed, that's what it suggests.  However, the code is for unconfirmed message, and the lower screenshot is from TTNv2 (exact same code, just different ini file), which shows the expected 5 uplinks and no downlinks.

**Comment by BNNorman** [comment]
Since my Kickstarter gaetway is defunct (at the moment) I can't do any checking with my end-node dev (coz they aren't being seen). I'll be looking for a single chan gateway to build to see if that helps.

Have you been able to view the downlink messages on the console IIRC you could see the data on V2.

**Comment by pjb304** [comment]
Alas my time to work on this is limited, and I ended up having to rebuilt a pi SD card last time I had put some time aside to debugging this so progress is slow - but yes looking in the console to see what the downlink messages are is on my todo list.  Thanks for the suggestion

**Comment by kersing** [comment]
Set the ‘verbose stream’ on in TTSCE and you’ll probably notice TTN is sending MAC commands downlink. These need to handled or you need to try to avoid them by setting device parameters. Check [TTN forum](https://www.thethingsnetwork.org/forum/t/stack-v3-sends-unrequested-downlink-data-for-new-abp-device/48327/13?u=kersing) for more information.

**Comment by pjb304** [comment]
Thanks @kersing for the hint. I think the messages that I'm seeing coming down are to do with ADR.  Unfortunately, the amount of time that I have available to work on this particularly bit of sofware has been diminishing for the last couple of years, and the deployment that we have been using this for is ending shortly so I can't justify the time to fix this, as all our devices currently using it will be decommissioned before TTN v2 is.

Sorry @BNNorman I know this will be a blow to your project, I wish you all the best with it.

**Comment by BNNorman** [comment]
No problem. I started working on handling MAC commands. I'll just have to continue with my variant. Thanks for the initial work you have done.
